### PR TITLE
Updates to azure support

### PR DESCRIPTION
**Pull request by SauarvD**
### Description

* `azure_enabled` should be `false` when extension is not applied
* Disabled API option for Azure

### CODE REVIEW

File: core/jazz_ui/src/app/pages/environment-detail/environment-detail.component.ts [hunk 1]
```diff
@@ -231,7 +231,11 @@ export class EnvironmentDetailComponent implements OnInit {
               return asset.asset_type === 'swagger_url';
             });
             if (foundAsset) {
-              return window.open(environment.urls['swagger_editor'] + foundAsset.provider_id);
+              if(foundAsset.provider == 'azure'){
```
Comment (raghits) [inline core/jazz_ui/src/app/pages/environment-detail/environment-detail.component.ts]: why this difference?

Comment (devsatishm) [reply core/jazz_ui/src/app/pages/environment-detail/environment-detail.component.ts]: Agreed, this experience should be consistent irrespective of the provider.

Comment (SauarvD) [reply core/jazz_ui/src/app/pages/environment-detail/environment-detail.component.ts]: reverted back the changes

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 2]
```diff
@@ -35,6 +35,7 @@ import groovy.transform.Field
 @Field def service_config
 @Field def websiteType
 @Field def azureAccount
+@Field def angularVersion
```
Comment (devsatishm) [inline builds/jenkins-build-pack-website/Jenkinsfile]: Defining this as a global var can affect builds on other user apps. Let's scope it to local by default.

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 3]
```diff
@@ -1211,7 +1213,11 @@ def buildWebsite(framework) {
 				sh "rm -rf dist"
 				sh "npm cache clean --force"
 				sh "npm install"
-				sh "ng build --output-path=dist"
+				if(angularVersion >= 6){
+					sh "ng build --output-path=dist"
+				} else {
+					sh "ng build --prod --output-path=dist"
```
Comment (devsatishm) [inline builds/jenkins-build-pack-website/Jenkinsfile]: Why is this always `--prod`? Shouldn't' this be environment specific?

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html [hunk 4]
```diff
@@ -20,9 +20,9 @@ <h4 class="bold-title">
                     <p class="step-title">Choose the type of service you want to create<span>*</span></p>
                 </div>
                 <div class="space-between content">
-                    <section class="pad-left0 service-box-section">
-                        <div tabindex="0" class="service-box" [ngClass]="{'service-box-selected': typeOfService == 'api'}" (keyup.enter)="changeServiceType('api')" (click)="changeServiceType('api')">
-                            <div class="icon-icon-api selected"></div>
+                    <section class="pad-left0 service-box-section disabled-box">
```
Comment (devsatishm) [inline core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html]: This should be disabled only when `provider = azure`.

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts [hunk 5]
```diff
@@ -396,7 +396,7 @@ export class CreateServiceComponent implements OnInit {
       this.typeOfPlatform = platformType;
       if(this.typeOfPlatform == 'azure'){
         if(this.typeOfService == 'sls-app'){
-          this.changeServiceType('api')
+          this.changeServiceType('function')
```
Comment (devsatishm) [inline core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: What's the purpose of this code? It is confusing since you are setting the service type as `function` when it is `sls-app`?

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts [hunk 6]
```diff
@@ -839,7 +839,7 @@ export class CreateServiceComponent implements OnInit {
     this.submitted = true;
     this.getData();
     this.createService();
-    this.typeOfService = 'api';
+    this.typeOfService = 'function';
```
Comment (devsatishm) [inline core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: What's the purpose of this change?

File: builds/jenkins-build-pack-website/Jenkinsfile (hunk 3 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: the command with the --prod flag was already present in the develop branch when I was merging it with azure support branch. And the reason for two different commands is to support new websites created with angular templates which are upgraded to 7 and to also support older angular websites of version 2.

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts (hunk 5 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: This was done on purpose based on edge case. If user selects sls-app service type and then selects AWS, it's fine. But if he wishes to change the platform from AWS to AZURE, the serviceType is still sls-app, which is not supported, that's why under this case, the serviceType is default to API, but since we are disabling api so by default it is function.

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts (hunk 6 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: So by default the service type is API, whenever you open the create-service window. Since we are disabling api now, so it changed to function. The way it is changed is wrong. It has to be this way ONLY if azure is the provider.

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html (hunk 4 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html]: Done

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts (hunk 6 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: Done

File: builds/jenkins-build-pack-website/Jenkinsfile (hunk 2 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: Done. Now passed as local variable

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 7]
```diff
@@ -200,7 +200,10 @@ node()  {
 			if(service_config['framework'] == 'angular' || service_config['framework'] == 'react') {
 				events.sendStartedEvent('BUILD', 'build starts', context_map, environment_logical_id)
 				try {
-					buildWebsite(service_config['framework'])
+					def packageJson = readFile("app/package.json")
+					def angularVersionData = new groovy.json.JsonSlurperClassic().parseText(packageJson).dependencies.'@angular/core'
+					def version = angularVersionData.substring(1, angularVersionData.indexOf('.'))
```
Comment (raghits) [inline builds/jenkins-build-pack-website/Jenkinsfile]: What happens if the build agent i.e. global angular cli has a different version of angular cli? Isn't that what counts here as we are using `ng build` for the build which uses global angular cli?

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts (hunk 5 above)
Comment (raghits) [reply core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: This needs to be cleaned up. If we do just top down filtering of options rather than going both ways.

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts [hunk 8]
```diff
@@ -164,6 +164,7 @@ export class CreateServiceComponent implements OnInit {
   public apigeeFeature = this.buildEnvironment.INSTALLER_VARS.feature.apigee && this.buildEnvironment.INSTALLER_VARS.feature.apigee.toString() === "true" ? true : false;
   public selectedDeploymentTarget = "aws_apigateway";
   public azureEnabled: boolean = false;
+  public disableFunction: boolean = false;
```
Comment (raghits) [inline core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: what is the purpose of this variable?

Comment (SauarvD) [reply core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.ts]: since there was no parameter to detect whether functions with NONE are selected or not and now we are NOT creating functions with NONE, that's why this variable is being introduced to take care of that scenario.

File: builds/jenkins-build-pack-website/Jenkinsfile (hunk 7 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: yes but in order to support the older services with the same build pack, if else condition was introduced for building older and newer angular versions of websites.

Comment (raghits) [reply builds/jenkins-build-pack-website/Jenkinsfile]: Now, if the angular cli in the package.json is 5.x, the angular cli in global is 7.x, u will be running with a new version of angular in global along with the --prod flag which will fail. Right?
